services/cloudaccounts/k8s/k8s.go

```diff
@@ -0,0 +1,101 @@
+package k8s
+
+import (
+	"fmt"
+	"github.com/dome9/dome9-sdk-go/services/cloudaccounts"
+	"net/http"
+	"time"
+)
+
+type CloudAccountRequest struct {
+	Name                   string                  `json:"name,omitempty"`
+	OrganizationalUnitID   string                  `json:"organizationalUnitId,omitempty"`
+}
+
+type CloudAccountResponse struct {
+	ID                     string                  `json:"id"`
+	Name                   string                  `json:"name"`
+	CreationDate           time.Time               `json:"creationDate"`
+	Vendor                 string                  `json:"vendor"`
+	OrganizationalUnitID   string                  `json:"organizationalUnitId,omitempty"`
+	OrganizationalUnitPath string                  `json:"organizationalUnitPath,omitempty"`
+	OrganizationalUnitName string                  `json:"organizationalUnitName,omitempty"`
+	ClusterVersion         string                  `json:"clusterVersion"`
+}
+
+type CloudAccountUpdateNameRequest struct {
+	Name                   string                  `json:"name"`
+}
+
+type CloudAccountUpdateOrganizationalIDRequest struct {
+	OrganizationalUnitId   string                  `json:"organizationalUnitId,omitempty"`
+}
+
+func (service *Service) Create(body CloudAccountRequest) (*CloudAccountResponse, *http.Response, error) {
+	v := new(CloudAccountResponse)
+	resp, err := service.Client.NewRequestDo("POST", cloudaccounts.RESTfulPathK8S, nil, body, v)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return v, resp, nil
+}
+
+func (service *Service) Get(id string) (*CloudAccountResponse, *http.Response, error) {
+	if id == "" {
+		return nil, nil, fmt.Errorf("id parameter is empty")
+	}
+
+	v := new(CloudAccountResponse)
+	relativeURL := fmt.Sprintf("%s/%s", cloudaccounts.RESTfulPathK8S, id)
+	resp, err := service.Client.NewRequestDo("GET", relativeURL, nil, nil, v)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return v, resp, nil
+}
+
+func (service *Service) Delete(id string) (*http.Response, error) {
+	if id == "" {
```

Comment:
why do you check if id is null? (the same for get and update)
we will just get error from backend if id is null.
I don't think there should be logic here that does not happen in the backend

Reply:
I did it because when an empty id is sent to the 'get' for example, The BE returns 200 OK with an empty body and then the http client tries to parse it as json and throws an error `panic: json: cannot unmarshal array into Go value of type k8s.CloudAccountResponse`
In other services they do verify that the 'options' parameter exists but only in the 'get' methods.
I will omit this validation and we will fix it on the BE side